compiler/rustc_codegen_ssa/src/mir/mod.rs

```diff
@@ -227,6 +230,13 @@ pub fn codegen_mir<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
             let layout = start_bx.layout_of(fx.monomorphize(decl.ty));
             assert!(!layout.ty.has_erasable_regions());
 
+            if layout.size.bytes() >= MIN_DANGEROUS_SIZE {
+                let (size_quantity, size_unit) = human_readable_bytes(layout.size.bytes());
+                cx.tcx().dcx().emit_warn(errors::DangerousStackAllocation {
```

Comment:
Is there *any* way we can associate this with a span? Emitting this error in a particularly large program gives us basically no information about where the fix needs to be applied, for example.

Reply:
Not sure where I should move this check- Asked in Zulip: https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/Issue.2083060.20-.20Regression.20with.20large.20stack.20arrays.20.282-4GB.29

I would appreciate some pointers. Thank you

Reply:
`decl.source_info.span` should give you the span you want to attach this warning to

Reply:
And based on my understanding of https://rustc-dev-guide.rust-lang.org/diagnostics/diagnostic-structs.html , you should be able to add that span to your `DangerousStackAllocation` struct, putting the `#[primary_span]` attribute onto it so that the diagnostic machinery will then attach that span to the message it prints.

Reply:
I will have a pull request soon! Thank you for getting back

Reply:
Made the suggested changes, but llvm-18 is not producing the expected warning and CI fails! How should that the addressed? I believe an associated PR was upstreamed to LLVM?

`failures:
  
  ---- [ui] tests/ui/codegen/large-stack-size-issue-83060.rs stdout ----
  diff of stderr:
  
  -	warning: Dangerous stack allocation of size: 1 GiB exceeds most architecture limits
  -	  --> $DIR/large-stack-size-issue-83060.rs:6:9
  -	   |
  -	LL |     let mut x: [u8; CAP>>1] = [0; CAP>>1];
  -	   |         ^^^^^
  -	
  -	warning: 1 warning emitted
  -	
  -	
  
  
  The actual stderr differed from the expected stderr.
  Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/codegen/large-stack-size-issue-83060/large-stack-size-issue-83060.stderr
  To update references, rerun the tests and pass the `--bless` flag
  To only update this specific test, also pass `--test-args codegen/large-stack-size-issue-83060.rs`
  
  error: 1 errors occurred comparing output.
  status: exit status: 0`

Reply:
@Dylan-DPC I need some help in knowing how to make the warning show up in CI.

Reply:
@rustbot review